odoo/cli/upgrade_code.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python3
+import argparse
+import sys
+
+import os.path
+from importlib.machinery import SourceFileLoader
+from pathlib import Path
+from typing import Iterator
+from types import ModuleType
+
+from . import Command
+
+"""
+Script method is add into the /upgrade/code directory.
+Every script should be in a file corresponding to his upgrade version (script for the )
+
+"""
+
+AVAILABLE_EXT = ('py', 'js', 'css', 'scss', 'xml', 'csv')
+
+
+class FileAccessor:
+    addon: Path
+    path: Path
+    content: str
+
+    def __init__(self, path: Path, addon: Path) -> None:
+        self.path = path
+        self.addon = addon
+        self._content = None
+        self.dirty = False
+
+    @property
+    def content(self):
+        if self._content is None:
+            with self.path.open("r") as f:
+                self._content = f.read()
+        return self._content
+
+    @content.setter
+    def content(self, value):
+        if self._content != value:
+            self._content = value
+            self.dirty = True
+
+
+class FileManager:
+    addons_path: list[str]
+    glob: str
+
+    def __init__(self, addons_path: list[str], glob: str = '**/*') -> None:
+        self.addons_path = addons_path
+        self.glob = glob
+        self._files = {
+            str(path): FileAccessor(path, addon)
+            for addon_path in addons_path
+            for addon in Path(addon_path).iterdir()
+            for path in addon.glob(glob)
+            if path.is_file() and path.name.rsplit('.')[-1] in AVAILABLE_EXT
```

Comment:
There's an API for getting a file extension from its `Path`:
```suggestion
            if path.is_file() and path.suffix in AVAILABLE_EXT
```